gpkit/nomials/variables.py

```diff
@@ -164,6 +165,14 @@ def __new__(cls, shape, *args, **descr):
             vl[i] = Variable(**descr)
             vl[i].key.arraykey = arraykey
 
+        if descr.pop("newvariable", True):
+            from .. import MODELS, MODELNUMS, NAMEDVARS
```

Comment:
The latter was the right thing to do! `NAMEDVARS` gets added to during Variable creation, which happens during VectorVariable creation...